### ISSUE TITLE
1.3.0 breaks compatiblity with `deno test` again

### ISSUE THREAD

**Issue by bjesuiter** [deno-library/progress#15]
The 'graceful handling of interruptions' in the following commit 

https://github.com/deno-library/progress/commit/c7f01b296bae7420215b9b256d4426bcc74660a8

breaks `deno test` by leaking async ops: 

<img width="1296" alt="Bildschirmfoto 2022-11-07 um 22 57 03" src="https://user-images.githubusercontent.com/2365676/200423933-04efa8ab-00fd-442d-bb8d-97c46ecf3dcd.png">

I'm currently looking into it how to fix it.

**Comment by bjesuiter** [comment]
I found the reason: 



## Broken
For some reason, binding the signalListener as a class method will leak async operations `deno test` as in my last screenshot. 

```ts
Deno.addSignalListener("SIGINT", this.signalListener.bind(this));
```

## Working 
Converting the class method to a lambda member in the class fixes the issue: 

```ts
class ProgressBar {
   private signalListener = () => {
      this.end();
      Deno.exit();
    };

  constructor() {
    Deno.addSignalListener("SIGINT", this.signalListener);
  }
}
```

Found that in the docs at https://deno.land/manual@v1.27.1/examples/os_signals (They only use the lambda version in the examples)

**Comment by fuxingZhang** [comment]
Landed in [v1.3.1](https://github.com/deno-library/progress/releases/tag/v1.3.1)

**Comment by fuxingZhang** [comment]
https://github.com/deno-library/progress/pull/16#issuecomment-1306450903
